topi/python/topi/cuda/batch_matmul.py

```diff
@@ -51,55 +62,73 @@ def _schedule(op):
             C = s.outputs[0].output(0)
 
         b, y, x = s[C].op.axis
-        y_bn = get_max_power2_factor(M, 64)
-        x_bn = get_max_power2_factor(N, 64)
-        by, y = s[C].split(y, y_bn)
-        bx, x = s[C].split(x, x_bn)
-        y_nthreads = min(y_bn, 8)
-        x_nthreads = min(x_bn, 8)
-        ty, yi = s[C].split(y, nparts=y_nthreads)
-        tx, xi = s[C].split(x, nparts=x_nthreads)
-        thread_x = te.thread_axis((0, x_nthreads), "threadIdx.x")
-        thread_y = te.thread_axis((0, y_nthreads), "threadIdx.y")
+
+        cfg.define_split("tile_y", y, num_outputs=3)
+        cfg.define_split("tile_x", x, num_outputs=3)
+        cfg.define_knob("auto_unroll_max_step", [8, 16, 32, 64])
+        target = tvm.target.Target.current()
+        if target.target_name in ['nvptx', 'rocm']:
+            # llvm-based backends cannot do non-explicit unrolling
+            cfg.define_knob("unroll_explicit", [1])
+        else:
+            cfg.define_knob("unroll_explicit", [0, 1])
+
+        if cfg.is_fallback:
```

Comment:
Missing `unroll_explicit` in fallback schedule. Based on your comment, this might result in a problem at LLVM-based backends.

Reply:
Good catch, thank you!

Reply:
Upon looking at this, I should use unroll explicit, not just define it.
What I'm unsure about is whether I need to define it in the fallback - conv2d_direct doesn't define it and it seems to work well.

Reply:
Hmm it's interesting. It means this parameter is never effective in those schedules...others like conv2d_direct uses it like:

`s[output].pragma(kernel_scope, 'unroll_explicit', cfg['unroll_explicit'].val)`

I think you can also fix them if you prefer to.

Reply:
I think it should be OK now.

Reply:
I think you still need to add `"unroll_explicit"` in the fallback cfg.

Reply:
Sorry just found that the `unroll_explicit` is still missing in fallback. Also would you mind moving L103 (define `tile_k`) up together with other tuning parameters?

Reply:
As much fun as geeking out over this is
- I don't think I need the fallback because
  - This schedule works without,
  - other schedules work without,
  - define_knob sets the fallback to the first option by setting `cfg._entity_map` which is what is queried by `__getitem__`.

I cannot move the definition of `tile_k` to before k is defined.

Reply:
If the first two points matter, we should just remove that parameter to make the tuning space more efficient. Meanwhile, I appreciate the third point that I didn't realize before.

For `tile_k`, I think you should be able to move the definition of `k` up as well. It should be safe because the tuning parameter must be static so it won't depend on other parameters. In this way, we can also put the fallback configs together to make it clearer.

Reply:
I moved the k and `tile_k` it appears to work and looks much prettier indeed. Thank you for insisting on it.

Reply:
What else do you think you need to happen?